apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/Logging/LoggerOptions.cs

```diff
@@ -103,14 +119,87 @@ public sealed class LoggerOptions
+            LogLevel logLevel = LogLevel.Information,
+            string logName = null,
+            Dictionary<string, string> labels = null,
+            MonitoredResource monitoredResource = null,
+            BufferOptions bufferOptions = null,
+            RetryOptions retryOptions = null,
+            TextWriter loggerDiagnosticsOutput = null) =>
+            CreateWithServiceContext(
+                logLevel,
+                logName,
+                labels,
+                monitoredResource,
+                bufferOptions,
+                retryOptions,
+                loggerDiagnosticsOutput,
+                Project.GetServiceName(null, monitoredResource),
+                Project.GetServiceVersion(null, monitoredResource));
+
+        /// <summary>
+        /// Create a new instance of <see cref="LoggerOptions"/>.
+        /// </summary>
+        /// <param name="logLevel">Optional, the minimum log level.  Defaults to <see cref="LogLevel.Information"/></param>
+        /// <param name="logName">Optional, the name of the log.  Defaults to 'aspnetcore'.</param>
+        /// <param name="labels">Optional, custom labels to be added to log entries.
+        /// Keys and values added to <paramref name="labels"/> should not be null.
+        /// If they are, an exception will be throw when attempting to log an entry.
+        /// The entry won't be logged and the exception will be propagated depending
+        /// on the value of <see cref="RetryOptions.ExceptionHandling"/>.</param>
+        /// <param name="monitoredResource">Optional, the monitored resource.  The monitored resource will
+        /// be automatically detected if it is not set and will default to the global resource if the detection fails.
+        /// See: https://cloud.google.com/logging/docs/api/v2/resource-list </param>
+        /// <param name="bufferOptions">Optional, the buffer options.  Defaults to a <see cref="BufferType.Timed"/></param>
+        /// <param name="retryOptions">Optional, the retry options.  Defaults to a <see cref="RetryType.None"/></param>
+        /// <param name="loggerDiagnosticsOutput">Optional. If set some logger diagnostics info will be written
+        /// to the given <see cref="TextWriter"/>. Currently the only diagnostics info we provide is the URL where
+        /// the logs written with these options can be found.</param>
+        /// <param name="serviceName">A string that represents the version of the service or the source code that logs are written for.
+        /// When set, it will be included in the <code>serviceContext</code> field of the log entry JSON payload.</param>
+        /// <param name="version">A string that represents the version of the service or the source code that logs are written for.
+        /// When set, it will be included in the <code>serviceContext</code> field of the log entry JSON payload.</param>
+        public static LoggerOptions CreateWithServiceContext(
```

Comment:
I was expecting the user to be able to supply a `Struct` as the complete service context here. Is that something that might be useful? Can the service context have more than the name and version? I think it's worth working that out during this PR, so we don't end up with more overloads than we want later on.